File: API/models/annonce.go
package models

type Annonce struct {
	Id                 int     `json:"id"`
	Titre              string  `json:"titre"`
	Description        string  `json:"description"`
	Type               string  `json:"type"`
	Prix               float64 `json:"prix"`
	StatutValidation   string  `json:"statut_validation"`
	CodePostal         string  `json:"code_postal"`
	Ville              string  `json:"ville"`
	Prenom             string  `json:"prenom"`
	Nom                string  `json:"nom"`
	Categorie          string  `json:"categorie"`
	CreatedAt          string  `json:"created_at"`
	Etat               string  `json:"etat"`
	PoidsKg            float64 `json:"poids_kg"`
	Quantite           int     `json:"quantite"`
	CommissionPrelevee float64 `json:"commission_prelevee"`
	PhotoUrl           string  `json:"photo_url"`
	IdUser             int     `json:"id_user"`
	IdCategorie        int     `json:"id_categorie"`
	Image              string  `json:"image"`
	StatutVente        string  `json:"statut_vente"`
	IsSponsored        bool    `json:"is_sponsored"`
	PlanAbo            string  `json:"plan_abo"`
}

type Categorie struct {
	Id      int    `json:"id"`
	Libelle string `json:"libelle"`
}

File: MobileClient/app/src/main/java/com/example/upcycleconnectclient/models/Annonce.kt
package com.example.upcycleconnectclient.models

data class Annonce(
    val id: Int,
    val titre: String?,
    val description: String?,
    val type: String?,
    val prix: Double,
    val ville: String?,
    val etat: String?,
    val categorie: String?,
    val image: String?,
    val prenom: String?,
    val nom: String?,
    val statut_vente: String?,
    val is_sponsored: Boolean,
    val plan_abo: String?
)
